bin/forklift

```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/env python3
+
+# Needs at least python 3.4 or the pathlib library
+# Needs click
+# Needs yaml
+
+import subprocess
+from collections import namedtuple
+from copy import deepcopy
+from distutils.version import StrictVersion
+from pathlib import Path
+
+import click
+import yaml
+
+ROOT = (Path(__file__).parent / '..').absolute()
+VERSIONS = ROOT / 'config' / 'versions.yaml'
+PIPELINES = ROOT / 'pipelines'
+
+
+# From https://stackoverflow.com/a/20666342
+def merge(source, destination):
+    """
+    >>> a = {'first': {'all_rows': {'pass': 'dog', 'number': '1'} } }
+    >>> b = {'first': {'all_rows': {'fail': 'cat', 'number': '5'} } }
+    >>> merge(b, a) == {'first': {'all_rows': {'pass': 'dog', 'fail': 'cat', 'number': '5'}}}
+    True
+    """
+    for key, value in source.items():
+        if isinstance(value, dict):
+            # get node or create one
+            node = destination.setdefault(key, {})
+            merge(value, node)
+        else:
+            destination[key] = value
+
+    return destination
+
+
+# From https://stackoverflow.com/a/1061350
+class ComparableMixin(object):
+    def __eq__(self, other):
+        return not self < other and not other < self
+
+    def __ne__(self, other):
+        return self < other or other < self
+
+    def __gt__(self, other):
+        return other < self
+
+    def __ge__(self, other):
+        return not self < other
+
+    def __le__(self, other):
+        return not other < self
+
+
+class Host(namedtuple('Host', 'name box memory'), ComparableMixin):
+    def __str__(self):
+        return self.name
+
+    def __lt__(self, other):
+        return self.name < other.name
+
+
+class Installer(namedtuple('Installer', 'foreman katello puppet boxes'), ComparableMixin):
+    """
+    The configuration for a specific installer version
+
+    A wrapper around entries in config/versions.yaml
+    """
+    # TODO: cache this?
+    @property
```

Comment:
It changes the function so you can use it like a property. In this case call it like ```installer._version``` rather than ```installer._version()```. You can also use it to define a setter function so you can control that as well, but that doesn't really make sense in this case.